### ISSUE TITLE
When I sleep (by lid or by Sleep button), my laptop still warm and battery drains very fast (10% per hour)

### ISSUE THREAD

**Issue by khoa-nv** [bavariancake/XPS9570-macOS#62]
I just install Mojave 10.14.6 and use this EFI folder. Everything working fine except sleep. When my laptop is in sleep mode, the screen and the CPU fan are turn off but it's still warm on the underside of the laptop. The battery loses about 1% every 5 minutes. So what happened and what do I have to do to fix it?
Another question is how to extend the battery life? I note my laptop can use for 8 hours on windows with nvidia, but now the battery only for 4 - 5hours.
Thank you so much!

My laptop XPS 9570 - i7 8750H - FHD 1920x1080 - 16GB RAM - 512GB SSD

**Comment by jaromeyer** [comment]
yup, there is definitely something wrong. I have exactly the same specs and can leave the laptop in standby for a few days. have you checked the kernel_task usage? eventually, you are using the wrong layout-id for your soundcard. and what wifi card are you using?

**Comment by khoa-nv** [comment]
> yup, there is definitely something wrong. I have exactly the same specs and can leave the laptop in standby for a few days. have you checked the kernel_task usage? eventually, you are using the wrong layout-id for your soundcard. and what wifi card are you using?

Thank you for your reply,
I'm using layout-id 30 for my soundcard and I have been replaced killer 1535 with dw1560. Does layout-id affect sleep?

**Comment by attilagyorffy** [comment]
I'd also like to learn a bit more about the different factors that affect the sleep functionality. If anyone has any good walkthrough on the subject, please share. Cheers.

**Comment by khoa-nv** [comment]
My problem was solved after I delete 2 files `SSDT-DeepIdle.aml` and `SSDT-DeepIdle.dsl`. What are the 2 files doing?

**Comment by calbear92** [comment]
I tried deleting those 2 patches. When my Hack Pro went to sleep, it became non-responsive. I put them back in after I rebooted.

**Comment by khoa-nv** [comment]
> I tried deleting those 2 patches. When my Hack Pro went to sleep, it became non-responsive. I put them back in after I rebooted.

Did you patch `SSDT-UPRW`?

**Comment by Alex2804** [comment]
sleep is not working at all on my XPS (32gb ram, 1TB 970evo, i7 8750H, FHD, DW1830) can you upload your EFI folder anywhere @khoa-nv ?
